contracts/utils/Redeemer.sol

```diff
@@ -121,6 +122,50 @@ contract Redeemer {
         }
     }
 
+    /**
+     * @dev helper to redeem proposal and execute the multi-call
+     * It calls execute on the proposal if it is not yet executed.
+     * It tries to redeem reputation and stake from the GenesisProtocol.
+     * It tries to execute multi call to contracts from the GenericSchemeMultiCall
+     * This function does not emit events.
+     * A client should listen to GenesisProtocol and GenericSchemeMultiCall events
+     * to monitor redemption and execution operations.
+     * @param _genericSchemeMultiCall GenericSchemeMultiCall scheme
+     * @param _genesisProtocol genesisProtocol
+     * @param _proposalId the ID of the voting in the voting machine
+     * @param _beneficiary beneficiary
+     * @return gpRewards array
+     *          gpRewards[0] - stakerTokenAmount
+     *          gpRewards[1] - voterReputationAmount
+     *          gpRewards[2] - proposerReputationAmount
+     * @return gpDaoBountyReward array
+     *         gpDaoBountyReward[0] - staker dao bounty reward -
+     *                                will be zero for the case there is not enough tokens in avatar for the reward.
+     *         gpDaoBountyReward[1] - staker potential dao bounty reward.
+     * @return executed  bool true or false
+     * @return winningVote
+     *                   1 - executed or closed and the winning vote is YES
+     *                   2 - executed or closed and the winning vote is NO
+     */
+    function redeemGenericSchemeMultiCall(GenericSchemeMultiCall _genericSchemeMultiCall,
+                            GenesisProtocol _genesisProtocol,
+                            bytes32 _proposalId,
+                            address _beneficiary)
+        external
+        returns(uint[3] memory gpRewards,
+                uint[2] memory gpDaoBountyReward,
+                bool executed,
+                uint256 winningVote
+                )
+    {
+        bool callGenericSchemeMultiCall;
+        (gpRewards, gpDaoBountyReward, executed, winningVote, callGenericSchemeMultiCall) =
+        genesisProtocolRedeem(_genesisProtocol, _proposalId, _beneficiary);
+        if (callGenericSchemeMultiCall) {
+            _genericSchemeMultiCall.execute(_proposalId);
```

Comment:
need to check if it already been executed...

Reply:
But if it is the proposal won't exist anymore anyway...